File: LeetCode/Explore/TopInterview/Amazon/ArrayString/49_group_anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        mp = dict()
        ans = list()
        
        for str in strs:
          tmp = ''.join(sorted(str))
          mp.setdefault(tmp, []).append(str)
          
        for v in mp.values():
          ans.append(v)
          
        return ans

File: LeetCode/Explore/TopInterview/Meta/ArrayString/49_group_anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
      unordered_map<string, vector<string> > mp;
      vector<vector<string> > ans;
      for (auto &str : strs) {
        string tmp = str;
        sort(tmp.begin(), tmp.end());
        mp[tmp].push_back(str);
      }
      for (auto &m : mp) {
        ans.push_back(m.second);
      }
      return ans;
    }
};
